src/Package/Impl/Imaging/ImagesProvider.cs

```diff
@@ -21,8 +21,14 @@ namespace Microsoft.VisualStudio.R.Package.Imaging {
 
     [Export(typeof(IImagesProvider))]
     internal sealed class ImagesProvider : IImagesProvider {
-        private Dictionary<string, ImageMoniker> _monikerCache = new Dictionary<string, ImageMoniker>();
-        private Lazy<Dictionary<string, string>> _fileExtensionCache = Lazy.Create(() => CreateExtensionCache());
+        private readonly Dictionary<string, ImageMoniker> _monikerCache = new Dictionary<string, ImageMoniker>();
+        private readonly Lazy<Dictionary<string, string>> _fileExtensionCache = Lazy.Create(() => CreateExtensionCache());
+
+        private static IVsImageService2 _imageService;
+
+        public ImagesProvider() {
+            _imageService = _imageService ?? VsAppShell.Current.GetGlobalService<IVsImageService2>(typeof(SVsImageService));
```

Comment:
Because request for icons in remote file completion case comes from a background thread